roles/rhn-subscription/tasks/main.yml

```diff
@@ -14,5 +14,5 @@
   with_items: "{{ rhn_subscription.repos.disable }}"
 
 - name: enable desire redhat repositories
-  command: subscription-manager repos --enable={{ item }}
+  command: subscription-manager repos --enable={{ rhn_subscription.repos.enable | join(' --enable=') }}
   with_items: "{{ rhn_subscription.repos.enable }}"
```

Comment:
You can get rid of the with_items since you no longer need to loop through them all.